haystack/components/connectors/openapi_service.py

```diff
@@ -211,7 +213,9 @@ def _invoke_method(self, openapi_service: OpenAPI, method_invocation_descriptor:
                 .get("schema", {})
                 .get("properties", {})
             ):
-                method_call_params["data"][param_name] = invocation_arguments.pop(param_name, "")
+                param_value = invocation_arguments.pop(param_name, None)
```

Comment:
Same thing, no? We are dealing with a copy here...we can even log unresolved params at the end perhaps...

Reply:
If we log unresolved params at the end then using .pop makes sense. Otherwise let's stick to `.get(param_name)`, which also returns None by default if the key is not found. Comparing .get and .pop, both retrieve the value for the given key but pop additionally removes it from the dict. Let's not do the latter unless we have a reason to.